base-notebook/start-singleuser.sh

```diff
@@ -4,19 +4,35 @@
 
 set -e
 
-notebook_arg=""
-if [ -n "${NOTEBOOK_DIR:+x}" ]
-then
-    notebook_arg="--notebook-dir=${NOTEBOOK_DIR}"
+# set default ip to 0.0.0.0
+if [[ "$NOTEBOOK_ARGS $@" != *"--ip="* ]]; then
+  NOTEBOOK_ARGS="--ip=0.0.0.0 $NOTEBOOK_ARGS"
 fi
 
-. /usr/local/bin/start.sh jupyterhub-singleuser \
-  --port=${JPY_PORT:-8888} \
-  --ip=0.0.0.0 \
-  --user=$JPY_USER \
-  --cookie-name=$JPY_COOKIE_NAME \
-  --base-url=$JPY_BASE_URL \
-  --hub-prefix=$JPY_HUB_PREFIX \
-  --hub-api-url=$JPY_HUB_API_URL \
-  ${notebook_arg} \
-  $@
+# handle some deprecated environment variables
+# from DockerSpawner < 0.8.
+# These won't be passed from DockerSpawner 0.9,
+# so avoid specifying --arg=empty-string
+if [ ! -z "$NOTEBOOK_DIR" ]; then
+  NOTEBOOK_ARGS="--notebook-dir=$NOTEBOOK_DIR $NOTEBOOK_ARGS"
```

Comment:
Good call. That's the only one that's a path, so I think the rest should be okay.